examples/05_assets/main.rs

```diff
@@ -0,0 +1,217 @@
+//! Demonstrates several assets-related techniques, including writing a custom
+//! asset loader, and loading assets from various paths.
+
+extern crate amethyst;
+extern crate cgmath;
+
+use amethyst::{Application, Event, State, Trans, VirtualKeyCode, WindowEvent};
+use amethyst::asset_manager::{AssetLoader, AssetLoaderRaw, AssetManager, Assets, DirectoryStore};
+use amethyst::config::Element;
+use amethyst::ecs::World;
+use amethyst::ecs::components::{LocalTransform, Mesh, Texture, Transform};
+use amethyst::ecs::resources::{Camera, Projection, ScreenDimensions};
+use amethyst::gfx_device::DisplayConfig;
+use amethyst::renderer::{Layer, PointLight, Pipeline, VertexPosNormal};
+use amethyst::renderer::pass::{Clear, DrawShaded};
+use cgmath::{Deg, Euler, Quaternion};
+use std::env::set_var;
+use std::str;
+
+// Implement custom asset loader that reads files with a simple format of
+// 1 vertex and 1 normal per line, with coordinates separated by whitespace.
+struct CustomObj {
+    vertices: Vec<[f32; 3]>,
+    normals: Vec<[f32; 3]>,
+}
+
+impl AssetLoaderRaw for CustomObj {
+    fn from_raw(_: &Assets, data: &[u8]) -> Option<CustomObj> {
+        let data: String = str::from_utf8(data).unwrap().into();
+        let mut vertices = vec![];
+        let mut normals = vec![];
+
+        for line in data.split("\n") {
```

Comment:
As long as you're changing this, can this line be updated to `for line in data.lines() {`? There were some issues with Windows line endings, and the changes to `.gitattributes` should fix that, but it wouldn't hurt to be safe.